### PR TITLE
Add support for assigning and unassigning tasks

### LINKED ISSUE #172: Prompt message for tasks does not explain what task types there are
Prompt and error messages for inputting tasks have some examples but do not state what types there are although they mention task types

### LINKED ISSUE #173: Error messages for different task types does not specify what details are necessary for each task type
When inputting `create-task tk/deadline`, for example, error message simply states "Task description is incomplete. Expected format: tk/[task type] [task details]." and user may not know what the task details are.

### LINKED ISSUE #175: Deadline task message does not specify correct input format
Entering command `create-task tk/deadline /by ` gives error "Invalid deadline format. Usage: create-task tk/deadline [description] /by [date]", which does not specify date-format - this would be useful for users to know.

### PR DESCRIPTION
**Pull request by DanzaSeah**
To be added in next PR
Create field in task, List of person that the task is assigned to 
Mark/unmark commands

Other issues fixed:
Fixes #172 
Fixes #173 
Fixes #175

### CODE REVIEW

**Comment by tingxuanp** [comment]
It seems like the current sample data has tasks assigned to people who are not yet vendors, is it possible to assign tasks to vendors only in the sample data?
image

**Comment by DanzaSeah** [comment]
> It seems like the current sample data has tasks assigned to people who are not yet vendors, is it possible to assign tasks to vendors only in the sample data? image

Sure, i'll make the changes

**Review by dasha3412 — CHANGES_REQUESTED** [review]
LGTM, concern is adding functionality for when a vendor is unassigned, the task should be unassigned from that former vendor also (currently, task remains assigned I believe)

File: src/main/java/seedu/address/logic/commands/task/CreateTaskCommand.java [hunk 1]
```diff
@@ -20,13 +22,18 @@ public class CreateTaskCommand extends Command {
 
     public static final String COMMAND_KEYWORD = "ctask";
 
-    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Creates a task in the address book. \n"
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Creates one or more tasks in the address book. \n"
```
[the remark this answers is not in the document]
Comment (DanzaSeah) [reply src/main/java/seedu/address/logic/commands/task/CreateTaskCommand.java:25]: > concern is adding functionality for when a vendor is unassigned, the task should be unassigned from that former vendor
That's true, i'll create an issue and implement it in the next pr, thanks

**Review by HanB1n — COMMENTED** [review]
LGTM

**Pull request merged**